### PR TITLE
docker_utils fixes for docker workspace path and for base-dir arg

### PR DESCRIPTION
**Pull request by czeslaw-spyro**
Correct base-dir arg for command in docker.
Correct working path (/workspace) evaluation in docker.

### CODE REVIEW

File: scargo/docker_utils.py [hunk 1]
```diff
@@ -25,10 +25,18 @@ def run_scargo_again_in_docker(
     build_env = project_config.build_env
     if build_env != SCARGO_DOCKER_ENV or Path("/.dockerenv").exists():
         return
+
     relative_path = Path.cwd().relative_to(project_path)
-    path_in_docker = Path("/workspace", relative_path)
+    path_in_docker = PurePosixPath("/workspace", relative_path)
 
     cmd_args = sys.argv[1:]
+    try:
```
Comment (aaksenczuk) [inline scargo/docker_utils.py:33]: It's important to note that the code might throw an IndexError if -B or --base-dir is the last argument in the cmd_args list. The code does not currently check if idx + 1 is within the bounds of the list.
```
    cmd_args = sys.argv[1:]
    try:
        # Correct base-dir arg for command in docker
        for idx, val in enumerate(cmd_args[:-1]):
            if val in ("-B", "--base-dir"):
                cmd_args[idx + 1] = "."
    except IndexError as e:
        logger.error("Expected a value after -B/--base-dir but none was provided: %s", e)
    except Exception as e:
        logger.error("Unexpected error occurred: %s", e)
```

Comment (czeslaw-spyro) [reply scargo/docker_utils.py:33]: I thought that 'except Exception as e:' will cover that case too?

Comment (czeslaw-spyro) [reply scargo/docker_utils.py:33]: I checked that "except Exception ..." will catch it as well:
Unexpected error occurred: %s list assignment index out of range

Comment (czeslaw-spyro) [reply scargo/docker_utils.py:33]: There is new improvement I can do is to add break statement.

**Review by aaksenczuk — APPROVED** [review]

**Pull request merged**